### PR TITLE
Create adapter_deps.bzl for Mixer

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR separates out the core dependencies that Mixer uses to compile protos (primarily, adapter config protos) into a skylark set of repository rules.

The goal of this PR is to establish an easy-to-use rule in WORKSPACE files to import the same deps that Mixer is using internally. The intent is to enable adapter authors to quickly establish the bazel-ification necessary to mirror Mixer setup. This should enable adapter authors to test/build outside of mixer repo (in advance of mixer repo integration).

The set of dependencies was created based on a set of bazel queries to inspect various adapter config `BUILD`s.

Example:

```bash
$ bazel query 'deps("//adapter/serviceControl/config:go_default_library")' | grep -iv "go1_" | grep -iv "local_config" | sed -e "s/\/\/.*//g" | sort | uniq -c
   7
   7 @bazel_tools
  94 @com_github_gogo_protobuf
   4 @com_github_golang_glog
  13 @com_github_golang_protobuf
 430 @com_github_google_protobuf
  23 @io_bazel_rules_go
   2 @io_bazel_rules_go_toolchain
  47 @org_golang_google_grpc
  55 @org_golang_x_net
  22 @org_golang_x_text
   2 @org_pubref_rules_protobuf
```

**Release note**:

```release-note
Create skylark rules for core mixer proto dependencies.
```

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/1077)

### CODE REVIEW

**Comment by qiwzhang** [comment]
May be related.   Could you also create a mixer_mock_test.bzl  to load dependencies need to build mixer mock test?

**Comment by douglas-reid** [comment]
@ZackButcher I've updated to `adapter_author_deps.bzl` and `mixer_adapter_repositories`. PTAL.

**Comment by ZackButcher** [comment]
Nit: update release note in PR description.

File: adapter_author_deps.bzl [hunk 1]
```diff
@@ -0,0 +1,57 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_repository")
+load('@bazel_tools//tools/build_defs/repo:git.bzl', 'git_repository')
```
Comment (lizan) [inline adapter_author_deps.bzl]: No need to load `git_repository`, it can be called by `native.git_repository` instantly.

Comment (douglas-reid) [inline adapter_author_deps.bzl]: Ah, good point. Thanks!

**Comment by douglas-reid** [comment]
PTAL. I had to update the `bin/bazel_to_go.py` script to handle a new source for repo deps (something to keep in mind for future adapter authors) and fix a linter issue (unparam) in `list_test.go`

File: bin/bazel_to_go.py [hunk 2]
```diff
@@ -127,20 +132,26 @@ def bazel_to_vendor(WKSPC):
         print "WORKSPACE file not found in " + WKSPC
         print "prog BAZEL_WORKSPACE_DIR"
         return -1
+    mixer_adapter_repos = WKSPC + "/adapter_author_deps.bzl"
```
Comment (lizan) [inline bin/bazel_to_go.py]: No necessary in this PR, but this kind of parsing can be done by `bazel query`

Feed the output of `bazel query 'kind(go_repository, //external:*)' --output=build` to the AST here would be better :)

Comment (mandarjog) [inline bin/bazel_to_go.py]: oh wow...
This is amazing !

Comment (douglas-reid) [inline bin/bazel_to_go.py]: agreed that would better, but I didn't want to rewrite the whole thing in this PR.

**Comment by douglas-reid** [comment]
I really don't understand. This PR causes no issues on my system, but generates infinite linter errors on prow.

File: adapter/list/list_test.go [hunk 3]
```diff
@@ -63,9 +63,9 @@ func TestBasic(t *testing.T) {
 	}
 }
 
-func contains(s []string, e string) bool {
+func containsListEntryTemplate(s []string) bool {
```
Comment (geeknoid) [inline adapter/list/list_test.go]: Why this change?

File: bin/linters.sh [hunk 4]
```diff
@@ -82,6 +82,7 @@ go_metalinter() {
         --enable=vet\
         --enable=vetshadow\
         --skip=testdata\
+        --skip=vendor\
         --vendor\
         $PKGS
 }
```
Comment (geeknoid) [inline bin/linters.sh]: Should we add this .bzl file and others like it to the list of files over which we run buildifier? See the bottom of linters.sh.

File: adapter/list/list_test.go (hunk 3 above)
Comment (douglas-reid) [inline adapter/list/list_test.go]: linter error: the method was only ever called with one string as `e`. `unparam` check.

File: bin/linters.sh (hunk 4 above)
Comment (douglas-reid) [inline bin/linters.sh]: i could be mistaken, but I was under the impression that `buildifier` only worked on `BUILD` files and could not be run on skylark files.
